src/main.rs

```diff
@@ -211,17 +192,65 @@ impl CargoCommand {
             .collect::<io::Result<_>>()
             .unwrap();
 
-        (command.wait().unwrap(), messages)
+        (process.wait().unwrap(), messages)
+    }
+
+    /// Create the cargo build command, but don't execute it.
+    /// If there is no pre-built std detected in the sysroot, `build-std` is used.
+    fn make_cargo_build_command(&self) -> Command {
+        let rustflags = env::var("RUSTFLAGS").unwrap_or_default()
+            + &format!(" -L{}/libctru/lib -lctru", env::var("DEVKITPRO").unwrap());
+        let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+        let sysroot = Self::find_sysroot();
+        let mut command = Command::new(cargo);
+
+        if !sysroot.join("lib/rustlib/armv6k-nintendo-3ds").exists() {
+            eprintln!("No pre-built std found, using build-std");
+            command.arg("-Z").arg("build-std");
+        }
+
+        command
+            .env("RUSTFLAGS", rustflags)
+            .arg(&self.command)
+            .arg("--target")
+            .arg("armv6k-nintendo-3ds")
+            .arg("--message-format")
+            .arg(&self.message_format)
+            .args(&self.args)
+            .stdout(Stdio::piped())
+            .stdin(Stdio::inherit())
+            .stderr(Stdio::inherit());
+
+        command
+    }
+
+    /// Get the compiler's sysroot path
+    fn find_sysroot() -> PathBuf {
+        let sysroot = env::var("SYSROOT").ok().unwrap_or_else(|| {
```

Comment:
Is this a standard environment variable used by Cargo? I couldn't find any docs or anything related to it (except https://github.com/rust-lang/cargo/issues/7501), although I suppose it doesn't hurt to make this configurable somehow.

Reply:
It isn’t supposed to be a Cargo defined variable. I believe @AzureMarker just put this in so you could use a different sysroot even while using the unmodified compiler.

Reply:
I was looking around and noticed that RLS checks for the variable, so I decided to include it:
https://github.com/rust-dev-tools/rls-analysis/blob/fd9939247dae2dbe4c801af4c0c3f82035f88d49/src/loader.rs#L158

Reply:
I don't think rust-analyzer checks for it though, and I don't see it mentioned anywhere else really.